node/src/actors/chain_manager/mod.rs

```diff
@@ -1439,13 +1439,17 @@ impl ChainManager {
         let mut batch_succeeded = true;
         let mut num_processed_blocks = 0;
 
-        for block in blocks.iter() {
+        for (i, block) in blocks.iter().enumerate() {
             num_processed_blocks += 1;
 
             if let Err(e) = self.process_requested_block(ctx, block.clone()) {
                 log::error!("Error processing block: {}", e);
-                self.initialize_from_storage(ctx);
-                log::info!("Restored chain state from storage");
+                if i > 0 {
```

Comment:
We already have a counter variable:

```suggestion
                if num_processed_blocks > 1 {
```